dev_tools/utils/nxdl_utils.py

```diff
@@ -357,14 +326,9 @@ def find_definition_file(bc_name):
     """
     bc_filename = None
     for nxdl_folder in ["contributed_definitions", "base_classes", "applications"]:
-        if os.path.exists(
-            f"{get_nexus_definitions_path()}{os.sep}"
-            f"{nxdl_folder}{os.sep}{bc_name}.nxdl.xml"
-        ):
-            bc_filename = (
-                f"{get_nexus_definitions_path()}{os.sep}"
-                f"{nxdl_folder}{os.sep}{bc_name}.nxdl.xml"
-            )
+        nxdl_file = nexus_def_path / nxdl_folder / f"{bc_name}.nxdl.xml"
+        if nxdl_file.exists():
+            bc_filename = nexus_def_path / nxdl_folder / f"{bc_name}.nxdl.xml"
```

Comment:
Isn't nxdl_file a Path object?

Reply:
After a quick check, it mostly is a Path object. But if we provide the definitions path as an environment variable, it can be a string. Look at this function:

```Python
def get_nexus_definitions_path():
    """Check NEXUS_DEF_PATH variable.
    If it is empty, this function is filling it"""
    try:  # either given by sys env
        return os.environ["NEXUS_DEF_PATH"]
    except KeyError:  # or it should be available locally under the dir 'definitions'
        local_dir = Path(__file__).resolve().parent
        for _ in range(2):
            local_dir = local_dir.parent
        return local_dir```
        
This is fun...

Reply:
ouch.
So lets just make it a path always, no?

Reply:
DONE